Mathlib/CategoryTheory/Limits/Shapes/Types.lean

```diff
@@ -929,4 +931,87 @@ lemma pushoutCocone_inl_eq_inr_iff_of_isColimit {c : PushoutCocone f g} (hc : Is
 
 end Pushout
 
-end CategoryTheory.Limits.Types
+end Types
+
+section Multiequalizer
+
+variable (I : MulticospanIndex (Type u))
+
+/-- Given a multiequalizer diagram `I : MulticospanIndex (Type u)` in the category
+of types, this is a type which identifies to the sections of the
+functor `I.multicospan.sections`. -/
+@[ext]
+structure MulticospanIndex.sections where
+  /-- The data of an element in `I.left i` for each `i : I.L`. -/
+  val (i : I.L) : I.left i
+  property (r : I.R) : I.fst r (val _) = I.snd r (val _)
+
+/-- The bijection `I.sections ≃ I.multicospan.sections` when `I : MulticospanIndex (Type u)`
+is a multiequalizer diagram in the category of types. -/
+@[simps]
+def MulticospanIndex.sectionsEquiv :
+    I.sections ≃ I.multicospan.sections where
+  toFun s :=
+    { val := fun i ↦ match i with
+        | .left i => s.val i
+        | .right j => I.fst j (s.val _)
+      property := by
+        rintro _ _ (_|_|r)
+        · rfl
+        · rfl
+        · exact (s.property r).symm }
+  invFun s :=
+    { val := fun i ↦ s.val (.left i)
+      property := fun r ↦ (s.property (.fst r)).trans (s.property (.snd r)).symm }
```

Comment:
```suggestion
  toFun s :=
  { val := fun i ↦ match i with
      | .left i => s.val i
      | .right j => I.fst j (s.val _)
    property := by
      rintro _ _ (_|_|r)
      · rfl
      · rfl
      · exact (s.property r).symm }
  invFun s :=
  { val := fun i ↦ s.val (.left i)
    property := fun r ↦ (s.property (.fst r)).trans (s.property (.snd r)).symm }
```

Reply:
I don't know if we have a rule for this, feel free to ignore. I just think it looks slightly better this way.

Reply:
Thanks for the reviews! but I do not think that we have any precise rule for `{...}`!